marshmallow/fields.py

```diff
@@ -1133,41 +1133,55 @@ def _deserialize(self, value, attr, data):
 class Function(Field):
     """A field that takes the value returned by a function.
 
-    :param callable func: A callable from which to retrieve the value.
+    :param callable func: This argument is to be depreciated. It exists for
+        backwards compatiblity. Use serialize instead.
+    :param callable deserialize: A callable from which to retrieve the value.
+        The function must take a single argument ``value`` which is the value
+        to be deserialized. It can also optionally take a ``context`` argument,
+        which is a dictionary of context variables passed to the deserializer.
+        If no callable is provided then ```value``` will be passed through
+        unchanged.
+    :param callable serialize: A callable from which to retrieve the value.
         The function must take a single argument ``obj`` which is the object
         to be serialized. It can also optionally take a ``context`` argument,
         which is a dictionary of context variables passed to the serializer.
-    :param callable deserialize: Deserialization function that takes the value
-        to be deserialized as its only argument.
+        If no callable is provided then the ```load_only``` flag will be set
+        to True.
     """
     _CHECK_ATTRIBUTE = False
 
-    def __init__(self, func, deserialize=None, **kwargs):
-        super(Function, self).__init__(**kwargs)
-        self.func = utils.callable_or_raise(func)
-        if deserialize:
-            self.deserialize_func = utils.callable_or_raise(deserialize)
-        else:
-            self.deserialize_func = None
+    def __init__(self, func=None, deserialize=None, serialize=None, **kwargs):
+        if func:
+            warnings.warn('Argument func of fields.Function is being depreciated.'
+                          ' Use the serialize argument instead.')
+            serialize = func
+
+        super(Function, self).__init__(load_only=not serialize, **kwargs)
+        self.serialize_func = serialize and utils.callable_or_raise(serialize)
+        self.deserialize_func = deserialize and utils.callable_or_raise(deserialize)
 
     def _serialize(self, value, attr, obj):
         try:
-            if len(utils.get_func_args(self.func)) > 1:
-                if self.parent.context is None:
-                    msg = 'No context available for Function field {0!r}'.format(attr)
-                    raise ValidationError(msg)
-                return self.func(obj, self.parent.context)
-            else:
-                return self.func(obj)
+            return self._call_or_raise(self.serialize_func, obj, attr)
         except AttributeError:  # the object is not expected to have the attribute
             pass
         return missing_
 
-    def _deserialize(self, value, attr, data):
+    def _deserialize(self, value, attr, obj):
```

Comment:
Let's keep the argument name `data` here for consistency with the other fields and for backwards compat.